internal/mode/static/state/dataplane/configuration.go

```diff
@@ -986,10 +998,29 @@ func buildAuxiliarySecrets(
 	return auxSecrets
 }
 
+func buildNginxPlus(g *graph.Graph) NginxPlus {
+	nginxPlusSettings := NginxPlus{AllowedAddresses: []string{"127.0.0.1"}}
+
+	ngfProxy := g.NginxProxy
+	if ngfProxy != nil && ngfProxy.Source.Spec.NginxPlus != nil {
+		if ngfProxy.Source.Spec.NginxPlus.AllowedAddresses != nil {
+			addresses := make([]string, len(ngfProxy.Source.Spec.NginxPlus.AllowedAddresses))
```

Comment:
```suggestion
			addresses := make([]string, 0, len(ngfProxy.Source.Spec.NginxPlus.AllowedAddresses))
```

The third arg is the capacity which is what you want. Then you don't have to specify the index and can just append.